### PR TITLE
Esas/migrating to react router v6

### PR DESCRIPTION
**Pull request by esasova**
Dev Review rules:
- 3 PR max
- at the second feedback, prefer meet instead of github conversation

Definition of Done:
- [ ] Testing
- [ ] No error logs in web console (except iFrame errors)
- [ ] Create or Update documentation (README file for users/integrators)
- [ ] Identify breaking changes with [conventional commits](https://kapeli.com/cheat_sheets/Conventional_Commits.docset/Contents/Resources/Documents/index)
- [ ] Check the [test page](https://spaceport.cosmotech.com/confluence/pages/viewpage.action?spaceKey=QA&title=Azure+Sample+Webapp) in order to manually validate the feature
- [ ] Clean code (regroup configuration variable, typos, no console.log...)

### CODE REVIEW

File: public/locales/fr/translation.json [hunk 1]
```diff
@@ -445,6 +445,11 @@
         "validate": "Valider",
         "validated": "Validé",
         "disabledTooltip": "Veuillez sauver ou annuler les modifications en cours avant de modifier le statut de validation du scénario"
+      },
+      "redirectError": {
+        "title": "Mauvaise URL",
+        "detail": "Le scénario avec id {{id}} n'existe plus.",
+        "comment": "Vous avez été redirigé à la page Scénario par défaut."
```
Comment (csm-thu) [inline public/locales/fr/translation.json:452]: Slightly better phrasing:
- `Le scénario avec l'id {{id}} n'existe plus.` 
- `Vous avez été redirigé vers la page Scénario par défaut.`

File: src/AppLayout.js [hunk 2]
```diff
@@ -10,19 +10,19 @@ const TABS = [
     key: 'tabs.scenario.key',
     label: 'layouts.tabs.scenario.tab.title',
     to: '/scenario',
-    render: () => <Scenario /> // eslint-disable-line
+    render: <Scenario /> // eslint-disable-line
```
Comment (csm-thu) [inline src/AppLayout.js:13]: I don't remember why we disabled eslint on this line, could you check if it is still useful after your change ? Thank you ! :smile:

Comment (esasova) [reply src/AppLayout.js:13]: I've already tried, we still need it because "React' must be in scope when using JSX" :slightly_smiling_face:

Comment (csm-thu) [reply src/AppLayout.js:13]: Ok, thank you :)

File: src/Routes.js [hunk 3]
```diff
@@ -9,13 +9,16 @@ import { SignIn as SignInView, AccessDenied as AccessDeniedView } from './views'
...
     <Routes_>
-      {/*
-      <Route path={'/'} element={<Navigate to="/scenario" replace />}></Route>
-*/}
-      <Route path="/sign-in" element={authenticated === false ? <SignInView /> : <Navigate to="/scenario" />} />
+      <Route path={'/'} element={<Navigate to={previousUrl || '/scenario'} replace />}></Route>
+      <Route
+        path="/sign-in"
+        element={
+          authenticated === false ? <SignInView /> : <Navigate to={history.state.idx === 0 ? '/scenario' : -1} />
```
Comment (nborde-CSM) [inline src/Routes.js:19]: Indeed `authenticated === false` is more explicit and readable but it is little shorter to use `!authenticated` test condition.

**Review by nborde-CSM — COMMENTED** [review]
One remark but LGTM :+1:

File: src/theme/custom/cosmoLight/picture.js [hunk 4]
```diff
@@ -3,7 +3,7 @@
 
 // Set here the logo and the login wallpaper url
 export default {
-  darkLogo: 'theme/cosmotech_dark_logo.png',
+  darkLogo: '/theme/cosmotech_dark_logo.png',
   lightLogo: 'theme/cosmotech_light_logo.png',
```
Comment (csm-thu) [inline src/theme/custom/cosmoLight/picture.js:7]: The leading slash is missing in the new `lightLogo` path :)

**Review by csm-thu — APPROVED** [review]
LGTM :+1: 
Kudos for the hard work !

**Pull request closed without merging**